doc/developer-notes.md

```diff
@@ -14,6 +14,7 @@ gradually.
   - No indentation for public/protected/private or for namespaces.
   - No extra spaces inside parenthesis; don't do ( this )
   - No space after function names; one space after if, for and while.
+  - `++i` is preferred over `i++`.
```

Comment:
Nit: Sorry to nitpick this after merge, but I'd rather put it in the section **General C++** (line 253) and also mention that `i++` is fine for integers.

You can copy the rationale from the [cpp guide](https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement)

>
When the return value is ignored, the "pre" form (++i) is never less efficient than the "post" form (i++), and is often more efficient. This is because post-increment (or decrement) requires a copy of i to be made, which is the value of the expression. If i is an iterator or other non-scalar type, copying i could be expensive. Since the two types of increment behave the same when the value is ignored, why not just always pre-increment?